sopel/bot.py

```diff
@@ -587,32 +587,48 @@ def call_rule(
         sopel: 'SopelWrapper',
         trigger: Trigger,
     ) -> None:
+        nick = trigger.nick
+        context = trigger.sender
+        is_channel = context and not trigger.is_privmsg
```

Comment:
This interface looks weird when you know that `context` already knows if it's a channel (`and not context.is_nick()`). Is it worth suggesting a change? No. But it still made me stop and think.

Reply:
After double checking, I can use context instead of trigger here.